apps/jvm.go

```diff
@@ -45,7 +46,7 @@ func (r MetricsReceiverJVM) Pipelines() []otel.Pipeline {
 		r.Endpoint = defaultJVMEndpoint
 	}
 
-	jarPath, err := findJarPath()
+	jarPath, err := FindJarPath()
```

Comment:
You could do something like `FindInstallPath` and assume some kind of standard installation directory where you can find components, like the `jmx-receiver.jar`.

This would the be usable if future integrations need the same sort of "find my bundled file from the distribution".

Reply:
@jsuereth I pushed a change to test your idea, let me know how you feel about it.

Reply:
Yeah this is going to be tricky: by exporting FindInstallPath and faking that instead of FindJarPath, the tests no longer elide the path joins. Not sure what the best way around that is. It's desirable to be able to test Windows conf generation from Linux, but it's undesirable to have to duplicate all of our golden test data just to have both kinds of path separators.

Would it make sense to change `updateOrCompareGolden` to treat paths as being equal if they only differ by path separator? Or wrap filepath.Join and export/fake that too? Both feel hacky.

Reply:
Reverted for now. We can revisit this in a future refactor.